### ISSUE TITLE
Update – I2V High-Resolution Bug Still Present + Severe RAM Usage Issue after last update

### ISSUE THREAD

**Issue by dsienra** [Haoming02/sd-webui-forge-classic#258]
After the latest update, the fade-to-white issue on high-resolution I2V is still present, and now there is also a severe RAM consumption problem.

Observations

Using 1280×720 resolution with the same models and Lighting LoRA:

33 frames → normal video

49 frames → normal video

65 frames → video shows fade-to-white artifact

81 frames → fade-to-white becomes complete

→ The issue remains identical to previous builds — the video generation completes (no crash), but the output becomes a white frame sequence beyond ~50 frames.

New Issue – Excessive RAM Consumption

After the latest update, system RAM usage keeps increasing with each generation.

I have 96 GB of RAM, and after one generation the usage doesn’t release memory properly.

Attempting a second generation without restarting Forge Neo causes the RAM to fill completely (96 GB used) and OOM error.

VRAM usage with ups and downs during generation, but it doesn’t utilize my full 24 GB GPU VRAM capacity during frame generation.

To continue working, I must close and reopen the Forge Neo console after each generation to clear RAM.

Summary

Fade-to-white bug remains in high-res I2V beyond 49 frames.

Memory leak / RAM overuse appears after the latest update, requiring a restart between runs.

VRAM usage stable but underutilized during frame generation (never close to full load).

Memory usage during first generation:
<img width="1106" height="625" alt="Image" src="https://github.com/user-attachments/assets/8e63124b-f397-4ca7-a67f-b1dbc1c2095b" />

Memory usage after second  generation and OOM error:
<img width="1106" height="650" alt="Image" src="https://github.com/user-attachments/assets/211ab4f1-bd21-4a0c-b3b0-2636c58d5272" />

VRAM usage during generation:

<img width="1359" height="636" alt="Image" src="https://github.com/user-attachments/assets/a1472b42-832a-4179-9f8d-f74eea88f299" />

1280x720 49 frames generation video sample:

https://github.com/user-attachments/assets/f13c273d-a4a1-4548-a2c8-8456b4cd329c

1280x720 81 frames generation video samples same settings:

https://github.com/user-attachments/assets/dfb1b9b9-75dd-4a2a-9051-2f2234cd9f20

https://github.com/user-attachments/assets/830fba73-4de8-4a7c-9d3f-d17b8aab03da

man walking in a post apocalyptic city landscape <lora:Wan2.2-Lightning_I2V-A14B-4steps-lora_HIGH_fp16:1>
Steps: 4, Sampler: Euler, Schedule type: Simple, CFG scale: 1, Seed: 3642607794, Size: 1280x720, Model hash: 6122e79d55, Model: wan2.2_i2v_high_noise_14B_fp8_scaled, Denoising strength: 1, RNG: CPU, Style Selector Enabled: True, Style Selector Randomize: False, Style Selector Style: base, Lora hashes: "Wan2.2-Lightning_I2V-A14B-4steps-lora_HIGH_fp16: 216cce62ecb5", Refiner: wan2.2_i2v_low_noise_14B_fp8_scaled [5471a457b6], Refiner switch at: 0.9, Version: neo, Diffusion in Low Bits: Automatic (fp16 LoRA), Module 1: wan_2.1_vae, Module 2: umt5_xxl_fp8_e4m3fn_scaled

Time taken: 7 min. 56.9 sec.

A: 13.78 GB, R: 15.66 GB, Sys: 17.0/24 GB (70.8%)

**Comment by Haoming02** [comment]
> Fade-to-white bug remains in high-res I2V beyond 49 frames

Can you try the same settings on ComfyUI to see if this also happens?

**Comment by dsienra** [comment]
sure:

<img width="1319" height="569" alt="Image" src="https://github.com/user-attachments/assets/0eb07a29-632b-499a-a931-85872ce355e3" />

also much faster:

Requested to load WanTEModel
loaded completely 21784.8 6419.477203369141 True
Requested to load WanVAE
loaded completely 5565.522792816162 242.02829551696777 True
Warning: TAESD previews enabled, but could not find models/vae_approx/None
Requested to load WAN21
loaded partially 8533.879601482868 8533.077774047852 203
Patching comfy attention to use sageattn
Patching torch settings: torch.backends.cuda.matmul.allow_fp16_accumulation = True
100%|████████████████████████████████████████████████████████████████████████████████████| 2/2 [02:12<00:00, 66.44s/it]
Restoring initial comfy attention
Patching torch settings: torch.backends.cuda.matmul.allow_fp16_accumulation = False
Using scaled fp8: fp8 matrix mult: False, scale input: True
model weight dtype torch.float16, manual cast: None
model_type FLOW
Warning: TAESD previews enabled, but could not find models/vae_approx/None
Requested to load WAN21
loaded partially 8524.879601482868 8524.879524230957 204
Patching comfy attention to use sageattn
Patching torch settings: torch.backends.cuda.matmul.allow_fp16_accumulation = True
100%|████████████████████████████████████████████████████████████████████████████████████| 2/2 [02:14<00:00, 67.47s/it]

comfyui generation:

https://github.com/user-attachments/assets/f372e617-e92a-4c03-b5ff-6a5d397f93ea

**Comment by dsienra** [comment]
Although this works fine in ComfyUI, I really like WebUI Forge Neo for its simplicity, clean interface, and overall workflow — I’d prefer to keep using it once this issue is resolved.

**Comment by AIEXAAA** [comment]
The observed issue arises due to the large size of the model, which results in it being almost fully loaded into system RAM. 

When the Refiner is enabled and execution completes, both GPU VRAM and CPU RAM are not fully released, leaving residual tensors in memory. 

Over multiple generation cycles, this residual allocation can accumulate, leading to scenarios where VRAM becomes insufficient.

 In such cases, the system may fall back to non-dedicated memory, which degrades performance and increases the likelihood of memory overflow errors.

**Comment by AIEXAAA** [comment]
**Summary**

Through systematic code testing and experiments across different PyTorch versions, the memory leak has been traced to the PyTorch version itself.

**Testing Methodology**

1. Set a breakpoint immediately after image generation completed, manually deleted all local variables (including model, tensors, and optimizers), and observed that memory usage did not decrease afterward. This ruled out residual references in the user script as the primary cause.
2. Identified reports indicating potential memory leak issues in PyTorch 2.8.0. Subsequently upgraded to PyTorch 2.9.1 with CUDA 13.0, along with the corresponding flash-attention version 2.8.3. Note that xFormers was not installed, as no pre-built wheel supporting PyTorch 2.9.1 + CUDA 13.0 was available at the time of testing.
3. Under the previous configuration (PyTorch 2.8.0 + xFormers+ flash-attention,SageAttention was not used), out-of-memory (OOM) errors typically occurred after approximately three video generations. With the new configuration, the process can now complete multiple generations without triggering OOM (see attached figure).

![Image](https://github.com/user-attachments/assets/e5a840cf-c119-4e47-a066-ff9791181c52)

**Discussion**

It remains unclear whether the issue has been fully resolved by bug fixes in PyTorch 2.9.1, or whether the absence of xFormers in the current setup simply avoids a specific interaction that previously triggered the leak.  

Based on the current evidence, the memory leak appears to be largely independent of the application code and is primarily associated with the PyTorch (and possibly xFormers) stack. Further controlled experiments (e.g., testing PyTorch 2.9.1 with a compatible future xFormers build) would help isolate the exact root cause.

**Comment by AIEXAAA** [comment]
Regarding the difference between PyTorch 2.9.1 + CUDA 13.0 + Flash-Attention 2.8.3 and PyTorch 2.9.1 + CUDA 13.0 + Flash-Attention 2.8.3 + xFormers: After verification, neither configuration triggers out-of-memory (OOM) errors during the main model inference phase.

 However, a notable regression was observed after adding xFormers: the VAE decoding stage of the WAN model (likely due to a misjudgment in memory estimation within the library or pipeline) becomes significantly more prone to GPU memory overflow. 

As a result, the GPU memory budget (GPU Weights) must be reduced from approximately 15,300 MiB to 14,900 MiB to maintain stable VAE operation under the same batch and resolution settings. 

In summary, while the core transformer attention path remains memory-safe in both setups, enabling xFormers introduces unexpected additional memory pressure specifically during VAE decoding, necessitating a ~400 MiB downward adjustment of the allocated GPU memory limit.

**Comment by Haoming02** [comment]
Please give the [dev](https://github.com/Haoming02/sd-webui-forge-classic/pull/559) branch a try if possible~

**Comment by elonmaxhimself** [comment]
have the same issue, using runpod template runpod/pytorch:2.8.0-py3.11-cuda12.8.1-cudnn-devel-ubuntu22.04
i was always installing the latest version of the neo branch, the pods that were started a month ago dont have the issue, the new pods run out of RAM after few hours of usage. Same usage pattern, same models being used on all pods.

**Comment by AIEXAAA** [comment]
After the update, when using Refiner to switch models, the Virtual Memory Load immediately overflowed (it was originally supposed to stay around 75%,now 95%).

It didn’t throw any errors; it simply terminated due to a memory overflow.

**Comment by elonmaxhimself** [comment]
interestingly enough i tried reverting to the older version of the repo, but expereiencing the same issue, with each generation where models are switching RAM usage goes up

**Comment by AIEXAAA** [comment]
After upgrading to the latest version (2.12), the virtual memory load usage dropped to 83%. 

Upgrade is recommended.

**Comment by Haoming02** [comment]
Hello, please update to the latest commit, and give the option `Reload "state_dict" Only` in **Settings/Refiner** a try~

Cut down RAM usage by ~20 GB

**Comment by AIEXAAA** [comment]
> Hello, please update to the latest commit, and give the option in **Settings/Refiner** a try~`Reload "state_dict" Only`
> 
> Cut down RAM usage by ~20 GB

When using the FP8 model, the maximum virtual memory load rate in the latest version is about 79%, measured against 128GB.

After enabling "Reload state_dict Only," the maximum virtual memory load rate is about 59%.

On my device, the execution time shows no difference. I once again recommend updating to the latest commit.

**Comment by Haoming02** [comment]
I believe this one can ~~finally~~ be closed now~

<p align="center">
<img src="https://github.com/user-attachments/assets/639bcf6f-c585-43b5-990f-87998a5822f3" />
</p>

<br>

https://github.com/user-attachments/assets/3fe66ace-7a2e-41a4-b47e-58c95cbeb353
